lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py

```diff
@@ -0,0 +1,243 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_deployment_facts
+version_added: "2.8"
+short_description: Get Azure Deployment facts.
+description:
+    - Get facts of Azure Deployment.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    name:
+        description:
+            - The name of the deployment.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get instance of Deployment
+    azure_rm_deployment_facts:
+      resource_group: myResourceGroup
+      name: myDeployment
+'''
+
+RETURN = '''
+deployments:
+    description: A list of dictionaries containing facts for Artifact.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The identifier of the resource.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Resources/deployments/myD
+                     eployment"
+        resource_group:
+            description:
+                - Resource group name.
+            returned: always
+            sample: myResourceGroup
+        name:
+            description:
+                - Deployment name.
+            returned: always
+            sample: myDeployment
+        provisioning_state:
+            description:
+                - Provisioning state of the deployment.
+            returned: always
+            sample: Succeeded
+        template_link:
+            description:
+                - Link to the template.
+            returned: always
+            sample: "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/d01a5c06f4f1bc03a049ca17bbbd6e06d62657b3/101-vm-simple-linux/
+                     azuredeploy.json"
+        parameters:
+            description:
+                - Dictionary containing deployment parameters.
+            returned: always
+            type: complex
+        outputs:
+            description:
+                - Dictionary containing deployment outputs.
+            returned: always
+        output_resources:
+            description:
+                - List of resources.
+            returned: always
+            type: complex
+            contains:
+                id:
+                    description:
+                        - Resource id.
+                    returned: always
+                    type: str
+                    sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkI
+                             nterfaces/myNetworkInterface"
+                name:
```

Comment:
is the name and type redundant? if not, then resource group should also been added.

Reply:
it's just coming from the api, so I have added it. maybe it's not necessary at all.
anyway, the resource group will be the same as deployment resource group.
I believe no ARM deployment creates resources outside it's resource group.